doc/nrf/gs_modifying.rst

```diff
@@ -84,54 +93,115 @@ The following CMake commands can be managed by SES, if they target the ``app`` l
-   It is possible to change the default configuration for a library by changing the :file:`Kconfig` file of the library.
-   However, best practice is to override the configuration in the application configuration file :file:`prj.conf`.
+The :file:`zephyr.dts` file describes the entire hardware-related configuration of the system, while the header file provides defines that can be used to refer hardware configuration from the source code.
+
+For more detailed information, see Zephyr's :ref:`zephyr:dt-guide`.
+
+.. _configure_application_sw:
+
+Software-related configuration
+------------------------------
+
+.. ncs-include:: guides/build/index.rst
+   :docset: zephyr
+   :dedent: 3
+   :start-after: Kconfig
+   :end-before: Information from devicetree is available to Kconfig,
+
+Information from devicetree is available to Kconfig, through the functions defined in :file:`zephyr/scripts/kconfig/kconfigfunctions.py`.
+
+The single :file:`.config` file describes the entire software configuration of the constructed binary.
+
+For detailed information about configuration options, see :ref:`zephyr:application-kconfig` in the Zephyr documentation.
+
+Memory layout configuration
+---------------------------
+
+If enabled, the :ref:`partition_manager` specific to the |NCS| provides the memory layout definition.
+The layout is impacted by various elements, such as Kconfig configuration options or the presence of child images.
+The build system works to ensure that all required partitions are enabled and that the right size of each partition area is selected.
+If you want to provide a custom partition definition, specify the static configuration in the :file:`pm_static.yml` file.
+
+After the CMake run, the build system creates a single :file:`partitions.yml` file in the :file:`build` directory.
+This file describes the resulting memory layout.
+This process also creates a set of header files that provides defines, which can be used to refer to memory layout elements.
+
+Child images
+------------
+
+The |NCS| build system allows the application project to become a root for the sub-applications known in the |NCS| as :ref:`child images <ug_multi_image>`.
+The child images are system components that reside on the same hardware system.
+They either cooperate with the application or the application depends on them.
+Examples of child images are bootloaders, network cores, or security-related binaries.
+
+Each child image is a separate application.
+For information about how to configure child images, see :ref:`ug_multi_image_defining`.
 
 Changing the configuration temporarily
 ======================================
 
 When building your application, the different :file:`Kconfig` and :file:`*_defconfig` files and the :file:`prj.conf` file are merged together.
-The combined configuration is saved in a :file:`zephyr/.config` file in your build directory.
+The combined configuration is saved in a :file:`zephyr/.config` file in your :file:`build` directory.
```

Comment:
```suggestion
The resulting configuration is written to the file :file:`zephyr/.config` in the :file:`build` directory.
```

Shouldn't use the word "save" about files in the build directory.

Files in the build directory are inherently temporary and not saved.

Reply:
These two sentences make it sound like .config is a merge of the aforementioned files. Which is incorrect.

Perhaps we could write:

[...] are merged together and then processed by Kconfig. The resulting configuration [...]

Reply:
Interestingly enough.

The .config file is not deleted when

ninja clean

is invoked.

So the sentence below should say

"but it is deleted when you **pristinely** clean the build directory"

Reply:
> The .config file is not deleted when
> ninja clean
> is invoked.

Correct, which was why I wrote to @pdunaj here https://github.com/nrfconnect/sdk-nrf/pull/6121#discussion_r759119361
> wrong, the .config is kept when you re-run CMake, but will be lost if you remove the build folder / do a pristine build.

So better write that instead of clean, cause a regular clean is not sufficient.

Reply:
Edited pristinely :D